stdlib/mutex_m/mutex_m.rbs

```diff
@@ -0,0 +1,77 @@
+# # mutex_m.rb
+#
+# When 'mutex_m' is required, any object that extends or includes Mutex_m will
+# be treated like a Mutex.
+#
+# Start by requiring the standard library Mutex_m:
+#
+#     require "mutex_m.rb"
+#
+# From here you can extend an object with Mutex instance methods:
+#
+#     obj = Object.new
+#     obj.extend Mutex_m
+#
+# Or mixin Mutex_m into your module to your class inherit Mutex instance methods
+# --- remember to call super() in your class initialize method.
+#
+#     class Foo
+#       include Mutex_m
+#       def initialize
+#         # ...
+#         super()
+#       end
+#       # ...
+#     end
+#     obj = Foo.new
+#     # this obj can be handled like Mutex
+#
+module Mutex_m
+  def self.append_features: (Module cl) -> untyped
+
+  def self.define_aliases: (Module cl) -> untyped
+
+  def self.extend_object: (Object obj) -> untyped
+
+  public
+
+  def mu_extended: () -> untyped
```

Comment:
These four methods are internal API that are called when it is included/extended.
So I didn't add any tests for them.